the-collab-lab/tcl-19-smart-shopping-list
# Ju rb creating new list

## Description
This PR adds one new component for creating a new list:  `CreateList`

### No list exists
Inside the `CreateList` component, we created a const `newToken`, which gets its value - a suitable token - from the script `src/lib/tokens.js` due to an `onClick` event.
The `localStorage.setItem` syntax saves this randomly generated token to `localStorage`. Because the submit button is inside a NavLink, you will then be automatically directed to the list view after pushing the `Create list` button.

### List already exists
If a token is already saved in `localStorage`, the user automatically gets directed to the `list` view due to the `if-statement` in `app.js` we created.

More about `localStorage`:point_right: [here.](https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem):point_left:

## Related Issue
Closes #3 

## Acceptance Criteria    

✓ For users who do not already have a token/list, a button or link exists on the home screen that allows them to create a new list 

✓ Clicking the button/link generates a new token and saves it to localStorage  

✓ Once the token has been created and saved, the user is redirected to the "list" view   

✓ For users who do have a token already saved in localStorage, they should be shown/redirected to the "list" view



## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before
Here you can see the homepage with the new `Create List` button.
![Homepage  `Create List` Button](https://docs.google.com/uc?export=download&id=1-ou9M4uEXXXdWv8eJqlj5AWQJU_3MlW-)


### After
After clicking the button, you will be directed to the `list` view and a token has been generated. If a token already exists, you will be directed to this view immediately after starting the app.
![List view and created token](https://docs.google.com/uc?export=download&id=1kYEFF-NRf6C7PCj9DOvKyPqJij7KaOB3)



## Testing Steps / QA Criteria

- From your terminal, pull down this branch with `git pull origin ju-rb-creating-new-list` and check that branch out with `git checkout ju-rb-creating-new-list`
- Run the app in your browser with `npm start`
- Open your localStorage within your browser dev tools, then push the `Create list` button to get directed to the list
- Check your localStorage for the new token
- Try going back to the homepage, which shouldn't be possible now because of the existing token.
